pyanaconda/modules/network/nm_client.py

```diff
@@ -0,0 +1,667 @@
+        return None
+    config = team.get_config()
+    return config
+
+def get_device_name_from_network_data(nm_client, network_data, supported_devices, bootif):
+    """Get the device name from kickstart device specification.
+
+    Generally given by --device option. For vlans also --interfacename
+    and --vlanid comes into play.
+
+    Side effect: for vlan sets network_data.parent value from --device option
+
+    :param network_data: a kickstart device configuartion
+    :type network_data: kickstart NetworkData object
+    :param supported_devices: list of names of supported devices
+    :type supported_devices: list(str)
+    :param bootif: MAC addres of device to be used for
+                    --device=bootif specification
+    :type bootif: str
+    :returns: device name the configuration should be used for
+    :rtype: str
+    """
+    spec = network_data.device
+    device_name = ""
+    msg = ""
+
+    if not spec:
+        msg = "device specification missing"
+
+    # Specification by device name
+    if spec in supported_devices:
+        device_name = spec
+        msg = "existing device found"
+    # Specification by mac address
+    elif ':' in spec:
+        device_name = get_iface_from_hwaddr(nm_client, spec) or ""
+        msg = "existing device found"
+    # Specification by BOOTIF boot option
+    elif spec == 'bootif':
+        device_name = get_iface_from_hwaddr(nm_client, bootif) or ""
+        msg = "existing device for {} found".format(bootif)
+    # First device with carrier (sorted lexicographically)
+    elif spec == 'link':
+        for candidate_name in sorted(supported_devices):
+            device = nm_client.get_device_by_iface(candidate_name)
+            if device and device.get_carrier():
+                device_name = device.get_iface()
+                break
+
+    if device_name:
+        if device_name not in supported_devices:
+            msg = "{} device found is not supported".format(device_name)
+            device_name = ""
+    # Virtual devices don't have to exist
+    elif spec and any((network_data.vlanid,
+                       network_data.bondslaves,
+                       network_data.teamslaves,
+                       network_data.bridgeslaves)):
+        device_name = spec
+        msg = "virtual device does not exist, which is OK"
+
+    if network_data.vlanid:
+        network_data.parent = device_name
+        if network_data.interfacename:
+            device_name = network_data.interfacename
+            msg = "vlan device name specified by --interfacename"
+        else:
+            device_name = default_ks_vlan_interface_name(device_name, network_data.vlanid)
+            msg = "vlan device name inferred from parent and vlanid"
+
+    log.debug("kickstart specification --device=%s -> %s (%s)", spec, device_name, msg)
+    return device_name
+
+
+def add_connection_from_ksdata(nm_client, network_data, device_name, activate=False, ifname_option_values=None):
```

Comment:
Could you please split this almost 200 lines function into smaller ones. Especially the big `if-elif-else` parts.

Reply:
https://github.com/rhinstaller/anaconda/pull/1771/commits/254e57729500af6b992ebb803fb2c1beb5798ab4